docs/docs/learn/shards/coding-with-shards.md

```diff
@@ -0,0 +1,220 @@
+# Coding with Shards
+
+In this chapter, we will be learning how to code with Shards so that you can write your very own program!
+
+## Data Types
+
+A shard can take in data, process it, and output the results.
+
```

Comment:
I will make a note about the input/output types below the table of different data types to keep the above explanation succinct. 

!!! note
    The input and output of a shard can be made up of multiple data types. This is done by encapsulating the different types within a sequence. For example `["a string", 1.04, true]` is a sequence made up of a string, a float, and a boolean value.


Would the explanation above be correct?

Reply:
Sorry, I wasn't clear.

I meant that input and output can be of several types in the sense that a shard can accept more than one data type but will only take one. And also return different types depending on some parameters or depending on the input type.

For instance `UI.Image` can receive either an `Image` or a `Texture` and it will "pass through" its input. So if the input was an `Image`, it will return that same `Image`. But if it was a `Texture`, it will return a `Texture`.

So in a sense the input types of `UI.Image` are `[Image Texture]` and the same for the output types, even though only one will be used for a given instance of the shard.

Hope it's clearer.

Reply:
Ah got it. Would it be better then if I changed the note to:

!!! note
A shard can have multiple data types as its input and output. For example, the shard `Math.Add` can have its input and output as an `Int`, or it can have its input and output as a `Float`.